[DA] ListCardView의 응모View 변경

kimkyunghun3 opened the pull request:
# 개요
- 🔗  이슈링크 : #80 


## 작업 내용
ListCardView에서 하단의 버튼과 응모관련 라벨에 대한 로직 수정했습니다.
BaseStackView를 3개 뷰가 상속받아서 사용하는 방식으로 구현했습니다.

시간관련 로직은 추후 로직이 추가되면 바꾸도록 하겠습니다

### 사용법

```swift
// progress 상태 시
let listCard = DDIPListCardView(.progress)
listCard.setListCardApplyView(applyDate: "2022.02.1")

//apply 상태 시
let listCard = DDIPListCardView(.apply)
listCard.setListCardDeadlineView(buttonColor: .designSystem(.secondarySkyblue200), isHidden: false, buttonTitle: .apply, titleStatus: .result, leftTime: "16")

// complete 상태 시
let listCard = DDIPListCardView(.complete)
listCard.setListCardCompleteView(drawDate: "2020.12.2")


// 동일 코드
listCard.setName(name: "아이스 아메리카노")
listCard.setBrandName(brand: "스타벅스")
listCard.setExpirationDate(expirationDate: "2022.12.31")
listCard.setImageIcon(image: "swift")
listCard.setApplyViewer(viewer: "100")

```

### 스크린샷
| 응모중 | 전달완료 | 응모하기 |
|:---:|:---:|:---:|
|<img src ="https://user-images.githubusercontent.com/52434820/182096799-7e8205c7-78c6-43ef-892c-d0c14b45dd6e.png" width ="200">|<img src ="https://user-images.githubusercontent.com/52434820/182096813-6f682665-ab15-43e5-81f4-07aff5684ae6.png" width ="200">|<img src ="https://user-images.githubusercontent.com/52434820/182096826-466c9494-297d-4b68-8cef-f6649cb1b7cc.png" width ="200">|

kimkyunghun3 committed:
:recycle: ListCardViewStyle status 추가

# :tada: = 프로젝트 시작 (#)
# :sparkles: = create 파일 생성 시 (#)
# :bulb: = update 코드 및 UI 업데이트, 기능 구현 시 (#)
# :rocket: = release 배포 시 (#)
# :recycle: = refactor 리팩토링 시 (기능은 같은데 코드가 달라지는경우) (#)
# :memo: = 문서 작성 시 (#)
# :art: = 디자인 시스템 관련 내용 (Color / Asset) (#)
# :hammer: = 테스트 코드 작성 (#)
################

# 커밋 타입 : 커밋 제목 (계정명/저장소명#이슈번호) 형식으로 작성하며 제목은 최대 50글자 정도로만 입력
# 제목을 아랫줄에 작성, 제목 끝에 마침표 금지, 무엇을 했는지 명확하게 작성

kimkyunghun3 committed:
:bulb: ListCardApplyBaseView 구현

# :tada: = 프로젝트 시작 (#)
# :sparkles: = create 파일 생성 시 (#)
# :bulb: = update 코드 및 UI 업데이트, 기능 구현 시 (#)
# :rocket: = release 배포 시 (#)
# :recycle: = refactor 리팩토링 시 (기능은 같은데 코드가 달라지는경우) (#)
# :memo: = 문서 작성 시 (#)
# :art: = 디자인 시스템 관련 내용 (Color / Asset) (#)
# :hammer: = 테스트 코드 작성 (#)
################

# 커밋 타입 : 커밋 제목 (계정명/저장소명#이슈번호) 형식으로 작성하며 제목은 최대 50글자 정도로만 입력
# 제목을 아랫줄에 작성, 제목 끝에 마침표 금지, 무엇을 했는지 명확하게 작성

kimkyunghun3 committed:
:bulb: ListCardApplyView 구현

# :tada: = 프로젝트 시작 (#)
# :sparkles: = create 파일 생성 시 (#)
# :bulb: = update 코드 및 UI 업데이트, 기능 구현 시 (#)
# :rocket: = release 배포 시 (#)
# :recycle: = refactor 리팩토링 시 (기능은 같은데 코드가 달라지는경우) (#)
# :memo: = 문서 작성 시 (#)
# :art: = 디자인 시스템 관련 내용 (Color / Asset) (#)
# :hammer: = 테스트 코드 작성 (#)
################

# 커밋 타입 : 커밋 제목 (계정명/저장소명#이슈번호) 형식으로 작성하며 제목은 최대 50글자 정도로만 입력
# 제목을 아랫줄에 작성, 제목 끝에 마침표 금지, 무엇을 했는지 명확하게 작성

kimkyunghun3 committed:
:bulb: ListCardCompleteView 구현

# :tada: = 프로젝트 시작 (#)
# :sparkles: = create 파일 생성 시 (#)
# :bulb: = update 코드 및 UI 업데이트, 기능 구현 시 (#)
# :rocket: = release 배포 시 (#)
# :recycle: = refactor 리팩토링 시 (기능은 같은데 코드가 달라지는경우) (#)
# :memo: = 문서 작성 시 (#)
# :art: = 디자인 시스템 관련 내용 (Color / Asset) (#)
# :hammer: = 테스트 코드 작성 (#)
################

# 커밋 타입 : 커밋 제목 (계정명/저장소명#이슈번호) 형식으로 작성하며 제목은 최대 50글자 정도로만 입력
# 제목을 아랫줄에 작성, 제목 끝에 마침표 금지, 무엇을 했는지 명확하게 작성

kimkyunghun3 committed:
:bulb: ListCardDeadlineView 구현

# :tada: = 프로젝트 시작 (#)
# :sparkles: = create 파일 생성 시 (#)
# :bulb: = update 코드 및 UI 업데이트, 기능 구현 시 (#)
# :rocket: = release 배포 시 (#)
# :recycle: = refactor 리팩토링 시 (기능은 같은데 코드가 달라지는경우) (#)
# :memo: = 문서 작성 시 (#)
# :art: = 디자인 시스템 관련 내용 (Color / Asset) (#)
# :hammer: = 테스트 코드 작성 (#)
################

# 커밋 타입 : 커밋 제목 (계정명/저장소명#이슈번호) 형식으로 작성하며 제목은 최대 50글자 정도로만 입력
# 제목을 아랫줄에 작성, 제목 끝에 마침표 금지, 무엇을 했는지 명확하게 작성

kimkyunghun3 committed:
:recycle: ListCardView 응모 StackView 사용에 따른 로직 변경

# :tada: = 프로젝트 시작 (#)
# :sparkles: = create 파일 생성 시 (#)
# :bulb: = update 코드 및 UI 업데이트, 기능 구현 시 (#)
# :rocket: = release 배포 시 (#)
# :recycle: = refactor 리팩토링 시 (기능은 같은데 코드가 달라지는경우) (#)
# :memo: = 문서 작성 시 (#)
# :art: = 디자인 시스템 관련 내용 (Color / Asset) (#)
# :hammer: = 테스트 코드 작성 (#)
################

# 커밋 타입 : 커밋 제목 (계정명/저장소명#이슈번호) 형식으로 작성하며 제목은 최대 50글자 정도로만 입력
# 제목을 아랫줄에 작성, 제목 끝에 마침표 금지, 무엇을 했는지 명확하게 작성

szzang-kr commented on Projects/DesignSystem/Sources/Component/ListCard/DDIPListCardViewStyle.swift:
```diff
@@ -9,23 +9,36 @@
     public let expirationDate: String
     public let iconImage: String
     public let description: String
+    public let applyStatus: ApplyTitleStatus
```
이전에 드렸던 피드백과 마찬가지로  brand, name, expirationDate, iconImage 등 카드 리스트에 담기는 모든 정보들은 API를 통해 정보를 받아오고, 해당 정보들로 변경될 다음과 같은 상황이 발생할것 같습니다.

```swift

// at collectionViewCell
private let cardListView = DDipListCardApplyView

...

func update(information: InformationModel) {
    cardListView.brand = information.brand
    cardListView.name = information.name
    cardListView.expirationDate = information.expirationDate
}

```

해당 상황을 생각하셔서 컴포넌트의 정보들을 교체하는 방법이 마련되어야 할 것 같습니다 !

kimkyunghun3 replied:
```swift
// progress 상태 시
let listCard = DDIPListCardView(.progress)
listCard.setListCardApplyView(applyDate: "2022.02.1")

//apply 상태 시
let listCard = DDIPListCardView(.apply)
listCard.setListCardDeadlineView(buttonColor: .designSystem(.secondarySkyblue200), isHidden: false, buttonTitle: .apply, titleStatus: .result, leftTime: "16")

// complete 상태 시
let listCard = DDIPListCardView(.complete)
listCard.setListCardCompleteView(drawDate: "2020.12.2")


// 동일 코드
listCard.setName(name: "아이스 아메리카노")
listCard.setBrandName(brand: "스타벅스")
listCard.setExpirationDate(expirationDate: "2022.12.31")
listCard.setImageIcon(image: "swift")
listCard.setApplyViewer(viewer: "100")
```

각각 상태에 따라 호출을 따로 해서 초기화시 사용하지 않고 필요한 메서드를 호출해서 만들어줄 수 있습니다.
Date부분에서는 데이터포맷터는 추후 적용하겠습니다!

AhnSangHee commented on Projects/DesignSystem/Sources/Component/ListCard/DDIPListCardView.swift:
```diff
@@ -9,15 +9,30 @@
 import UIKit
 
 public class DDIPListCardView: UIView, AddViewsable {
```
추후 가능하실 때 DDIPListCardView에서 DDipListCardView로 네이밍 수정해주시면 감사하겠습니다!

kimkyunghun3 replied:
넵 이부분은 추후 한번에 변경하겠습니다~

AhnSangHee reviewed (changes requested):
@dvHuni 님이 코멘트 해주신 것만 수정해주시면 될 것 같아요! 고생하셨습니다 👍

kimkyunghun3 committed:
:recycle: ListCardViewStyle 파일 삭제

# :tada: = 프로젝트 시작 (#)
# :sparkles: = create 파일 생성 시 (#)
# :bulb: = update 코드 및 UI 업데이트, 기능 구현 시 (#)
# :rocket: = release 배포 시 (#)
# :recycle: = refactor 리팩토링 시 (기능은 같은데 코드가 달라지는경우) (#)
# :memo: = 문서 작성 시 (#)
# :art: = 디자인 시스템 관련 내용 (Color / Asset) (#)
# :hammer: = 테스트 코드 작성 (#)
################

# 커밋 타입 : 커밋 제목 (계정명/저장소명#이슈번호) 형식으로 작성하며 제목은 최대 50글자 정도로만 입력
# 제목을 아랫줄에 작성, 제목 끝에 마침표 금지, 무엇을 했는지 명확하게 작성

kimkyunghun3 committed:
:recycle: ApplyViewer 로직 수정

# :tada: = 프로젝트 시작 (#)
# :sparkles: = create 파일 생성 시 (#)
# :bulb: = update 코드 및 UI 업데이트, 기능 구현 시 (#)
# :rocket: = release 배포 시 (#)
# :recycle: = refactor 리팩토링 시 (기능은 같은데 코드가 달라지는경우) (#)
# :memo: = 문서 작성 시 (#)
# :art: = 디자인 시스템 관련 내용 (Color / Asset) (#)
# :hammer: = 테스트 코드 작성 (#)
################

# 커밋 타입 : 커밋 제목 (계정명/저장소명#이슈번호) 형식으로 작성하며 제목은 최대 50글자 정도로만 입력
# 제목을 아랫줄에 작성, 제목 끝에 마침표 금지, 무엇을 했는지 명확하게 작성

kimkyunghun3 committed:
:recycle: CardListButtonStyle 삭제

# :tada: = 프로젝트 시작 (#)
# :sparkles: = create 파일 생성 시 (#)
# :bulb: = update 코드 및 UI 업데이트, 기능 구현 시 (#)
# :rocket: = release 배포 시 (#)
# :recycle: = refactor 리팩토링 시 (기능은 같은데 코드가 달라지는경우) (#)
# :memo: = 문서 작성 시 (#)
# :art: = 디자인 시스템 관련 내용 (Color / Asset) (#)
# :hammer: = 테스트 코드 작성 (#)
################

# 커밋 타입 : 커밋 제목 (계정명/저장소명#이슈번호) 형식으로 작성하며 제목은 최대 50글자 정도로만 입력
# 제목을 아랫줄에 작성, 제목 끝에 마침표 금지, 무엇을 했는지 명확하게 작성

kimkyunghun3 committed:
:recycle: CardListButton 로직 변경

# :tada: = 프로젝트 시작 (#)
# :sparkles: = create 파일 생성 시 (#)
# :bulb: = update 코드 및 UI 업데이트, 기능 구현 시 (#)
# :rocket: = release 배포 시 (#)
# :recycle: = refactor 리팩토링 시 (기능은 같은데 코드가 달라지는경우) (#)
# :memo: = 문서 작성 시 (#)
# :art: = 디자인 시스템 관련 내용 (Color / Asset) (#)
# :hammer: = 테스트 코드 작성 (#)
################

# 커밋 타입 : 커밋 제목 (계정명/저장소명#이슈번호) 형식으로 작성하며 제목은 최대 50글자 정도로만 입력
# 제목을 아랫줄에 작성, 제목 끝에 마침표 금지, 무엇을 했는지 명확하게 작성

kimkyunghun3 committed:
:recycle: ListCardView 로직 변경

# :tada: = 프로젝트 시작 (#)
# :sparkles: = create 파일 생성 시 (#)
# :bulb: = update 코드 및 UI 업데이트, 기능 구현 시 (#)
# :rocket: = release 배포 시 (#)
# :recycle: = refactor 리팩토링 시 (기능은 같은데 코드가 달라지는경우) (#)
# :memo: = 문서 작성 시 (#)
# :art: = 디자인 시스템 관련 내용 (Color / Asset) (#)
# :hammer: = 테스트 코드 작성 (#)
################

# 커밋 타입 : 커밋 제목 (계정명/저장소명#이슈번호) 형식으로 작성하며 제목은 최대 50글자 정도로만 입력
# 제목을 아랫줄에 작성, 제목 끝에 마침표 금지, 무엇을 했는지 명확하게 작성

kimkyunghun3 committed:
:recycle: ListCardApplyBaseView 로직 변경

# :tada: = 프로젝트 시작 (#)
# :sparkles: = create 파일 생성 시 (#)
# :bulb: = update 코드 및 UI 업데이트, 기능 구현 시 (#)
# :rocket: = release 배포 시 (#)
# :recycle: = refactor 리팩토링 시 (기능은 같은데 코드가 달라지는경우) (#)
# :memo: = 문서 작성 시 (#)
# :art: = 디자인 시스템 관련 내용 (Color / Asset) (#)
# :hammer: = 테스트 코드 작성 (#)
################

# 커밋 타입 : 커밋 제목 (계정명/저장소명#이슈번호) 형식으로 작성하며 제목은 최대 50글자 정도로만 입력
# 제목을 아랫줄에 작성, 제목 끝에 마침표 금지, 무엇을 했는지 명확하게 작성

kimkyunghun3 committed:
:recycle: ListCardCompleteView 로직 변경

# :tada: = 프로젝트 시작 (#)
# :sparkles: = create 파일 생성 시 (#)
# :bulb: = update 코드 및 UI 업데이트, 기능 구현 시 (#)
# :rocket: = release 배포 시 (#)
# :recycle: = refactor 리팩토링 시 (기능은 같은데 코드가 달라지는경우) (#)
# :memo: = 문서 작성 시 (#)
# :art: = 디자인 시스템 관련 내용 (Color / Asset) (#)
# :hammer: = 테스트 코드 작성 (#)
################

# 커밋 타입 : 커밋 제목 (계정명/저장소명#이슈번호) 형식으로 작성하며 제목은 최대 50글자 정도로만 입력
# 제목을 아랫줄에 작성, 제목 끝에 마침표 금지, 무엇을 했는지 명확하게 작성

kimkyunghun3 committed:
:recycle: ListCardApplyView 로직 변경

# :tada: = 프로젝트 시작 (#)
# :sparkles: = create 파일 생성 시 (#)
# :bulb: = update 코드 및 UI 업데이트, 기능 구현 시 (#)
# :rocket: = release 배포 시 (#)
# :recycle: = refactor 리팩토링 시 (기능은 같은데 코드가 달라지는경우) (#)
# :memo: = 문서 작성 시 (#)
# :art: = 디자인 시스템 관련 내용 (Color / Asset) (#)
# :hammer: = 테스트 코드 작성 (#)
################

# 커밋 타입 : 커밋 제목 (계정명/저장소명#이슈번호) 형식으로 작성하며 제목은 최대 50글자 정도로만 입력
# 제목을 아랫줄에 작성, 제목 끝에 마침표 금지, 무엇을 했는지 명확하게 작성

kimkyunghun3 committed:
:recycle: ListCardDeadlineView 로직 변경

# :tada: = 프로젝트 시작 (#)
# :sparkles: = create 파일 생성 시 (#)
# :bulb: = update 코드 및 UI 업데이트, 기능 구현 시 (#)
# :rocket: = release 배포 시 (#)
# :recycle: = refactor 리팩토링 시 (기능은 같은데 코드가 달라지는경우) (#)
# :memo: = 문서 작성 시 (#)
# :art: = 디자인 시스템 관련 내용 (Color / Asset) (#)
# :hammer: = 테스트 코드 작성 (#)
################

# 커밋 타입 : 커밋 제목 (계정명/저장소명#이슈번호) 형식으로 작성하며 제목은 최대 50글자 정도로만 입력
# 제목을 아랫줄에 작성, 제목 끝에 마침표 금지, 무엇을 했는지 명확하게 작성

kimkyunghun3 committed:
:recycle: image custom asset 및 binding 방법 변경

# :tada: = 프로젝트 시작 (#)
# :sparkles: = create 파일 생성 시 (#)
# :bulb: = update 코드 및 UI 업데이트, 기능 구현 시 (#)
# :rocket: = release 배포 시 (#)
# :recycle: = refactor 리팩토링 시 (기능은 같은데 코드가 달라지는경우) (#)
# :memo: = 문서 작성 시 (#)
# :art: = 디자인 시스템 관련 내용 (Color / Asset) (#)
# :hammer: = 테스트 코드 작성 (#)
################

# 커밋 타입 : 커밋 제목 (계정명/저장소명#이슈번호) 형식으로 작성하며 제목은 최대 50글자 정도로만 입력
# 제목을 아랫줄에 작성, 제목 끝에 마침표 금지, 무엇을 했는지 명확하게 작성

szzang-kr commented on Projects/DesignSystem/Sources/Component/ApplyViewer/DDIPApplyViewer.swift:
```diff
@@ -27,20 +27,24 @@ public class DDIPApplyViewer: UIView {
+        super.init(frame: .zero)
         setLayout()
         setAttribute()
         applyViewerLabel.font = UIFont.systemFont(ofSize: 10)
+        applyViewerLabel.font = .designSystem(.pretendard, family: .regular, size: ._14)
```
setAttribute() method가 있는데, init에 넣은신 이유가 있을까요 ? 🤔

kimkyunghun3 replied:
이 부분은 폰트로 size 10이 디자인시스템에 없어서 일단 보이게 하려고 옮겨두긴했습니다! 그래도 init에 넣는건 적절하지 않으니 이동해두었습니다!

szzang-kr commented on Projects/DesignSystem/Sources/Component/ListCard/DDipListCardDeadlineView.swift:
```diff
@@ -0,0 +1,20 @@
+import UIKit
+
+public class DDipListCardDeadlineView: DDipListCardApplyBaseView {
+    public func setListCardButton(buttonTitle: DDIPCardListButton.TitleStatus, buttonColor: UIColor?, isHidden: Bool) {
+        cardListButton.setButtonAttribute(titleStatus: buttonTitle, buttonColor: buttonColor, isHidden: isHidden)
+    }
+
+    public func setDrawLabel(titleStatus: String, leftTime: String) {
+        drawLabel.text = "\(titleStatus)까지 \(leftTime)시간 남았어요!"
+        drawLabel.font = .designSystem(.pretendard, family: .regular, size: ._14)
+    }
```
💯 LGTM

szzang-kr commented on Projects/DesignSystem/Sources/Component/ListCard/DDipListCardApplyView.swift:
```diff
@@ -0,0 +1,18 @@
+    public func setDrawLabel(applyDate: String) {
+        drawLabel.text = "\(applyDate) 응모"
+        drawLabel.textColor = .designSystem(.neutralGray500)
+        drawLabel.font = .designSystem(.pretendard, family: .regular, size: ._14)
+        cardListButton.setButtonAttribute(titleStatus: .progress, buttonColor: .designSystem(.secondarySkyblue200), isHidden: false)
```
마찬가지로 변경되지 않는 내용은 따로 분리하면 좋을것 같습니다!

kimkyunghun3 replied:
요 부분과 나머지 View들도 아래의 커멘트 방식으로 모두 수정했습니다!

szzang-kr commented on Projects/DesignSystem/Sources/Component/ListCard/DDipListCardCompleteView.swift:
```diff
@@ -0,0 +1,19 @@
+public class DDipListCardCompleteView: DDipListCardApplyBaseView {
+    public func setDrawLabel(drawDate: String) {
+        drawLabel.text = "\(drawDate) 전달 완료"
+        drawLabel.font = .designSystem(.pretendard, family: .bold, size: ._14)
+        drawLabel.textColor = .designSystem(.secondaryBlue)
+
+        cardListButton.isHidden = true
```
drawDate를 통해 텍스트만 변경되는데, 변경되지 않는 값들은 init시점에 setAttribute() 같은 함수로 분리해두면 좋을것 같아요!

변경되지 않는 코드지만, 변경하는 메소드 안에 있어 변경될것 같은 느낌을 주네요 😄

kimkyunghun3 replied:
```swift
public class DDipListCardDeadlineView: DDipListCardApplyBaseView {
    public override init() {
        super.init()
        setAttribute()
    }

    required init(coder: NSCoder) {
        fatalError("init(coder:) has not been implemented")
    }

// .... 생략

    public func setAttribute() {
        drawLabel.font = .designSystem(.pretendard, family: .regular, size: ._14)
    }
```
고정되는 부분은 이렇게 분리해서 init 시점에 호출하도록 했습니다!

szzang-kr commented on Projects/DesignSystem/Sources/Component/ListCard/DDIPListCardView.swift:
```diff
@@ -57,64 +126,46 @@ public class DDIPListCardView: UIView, AddViewsable {
     }
     
     private func setView() {
+        self.backgroundColor = .designSystem(.neutralWhite)
+        self.layer.cornerRadius = 8
```
setView와 setAttribute의 차이는 무엇인가요? 🤔

내용을 보니까 비슷한 성격인것 같은데 분리하신 이유가 궁금합니다.

kimkyunghun3 replied:
저도 확인해보니 비슷한느낌이라서 setView는 말그대로 View를 settting해줄때(레이아웃 관련) 사용하려고 두었고, setAttribute는 속성들을 세팅해줄때로 변경했습니다!

szzang-kr reviewed (changes requested):
조금 무리한 일정이였는데 수고하셨습니다.. 😭

kimkyunghun3 committed:
:recycle: font 설정 메서드로 이동

# :tada: = 프로젝트 시작 (#)
# :sparkles: = create 파일 생성 시 (#)
# :bulb: = update 코드 및 UI 업데이트, 기능 구현 시 (#)
# :rocket: = release 배포 시 (#)
# :recycle: = refactor 리팩토링 시 (기능은 같은데 코드가 달라지는경우) (#)
# :memo: = 문서 작성 시 (#)
# :art: = 디자인 시스템 관련 내용 (Color / Asset) (#)
# :hammer: = 테스트 코드 작성 (#)
################

# 커밋 타입 : 커밋 제목 (계정명/저장소명#이슈번호) 형식으로 작성하며 제목은 최대 50글자 정도로만 입력
# 제목을 아랫줄에 작성, 제목 끝에 마침표 금지, 무엇을 했는지 명확하게 작성

kimkyunghun3 committed:
:recycle: 변하지않는 값들 메서드 분리 및 init에서 호출하도록 로직변경

# :tada: = 프로젝트 시작 (#)
# :sparkles: = create 파일 생성 시 (#)
# :bulb: = update 코드 및 UI 업데이트, 기능 구현 시 (#)
# :rocket: = release 배포 시 (#)
# :recycle: = refactor 리팩토링 시 (기능은 같은데 코드가 달라지는경우) (#)
# :memo: = 문서 작성 시 (#)
# :art: = 디자인 시스템 관련 내용 (Color / Asset) (#)
# :hammer: = 테스트 코드 작성 (#)
################

# 커밋 타입 : 커밋 제목 (계정명/저장소명#이슈번호) 형식으로 작성하며 제목은 최대 50글자 정도로만 입력
# 제목을 아랫줄에 작성, 제목 끝에 마침표 금지, 무엇을 했는지 명확하게 작성

kimkyunghun3 committed:
:recycle: setView, setAttribute 메서드 역할에 따른 내부 프로퍼티들 위치 변경

# :tada: = 프로젝트 시작 (#)
# :sparkles: = create 파일 생성 시 (#)
# :bulb: = update 코드 및 UI 업데이트, 기능 구현 시 (#)
# :rocket: = release 배포 시 (#)
# :recycle: = refactor 리팩토링 시 (기능은 같은데 코드가 달라지는경우) (#)
# :memo: = 문서 작성 시 (#)
# :art: = 디자인 시스템 관련 내용 (Color / Asset) (#)
# :hammer: = 테스트 코드 작성 (#)
################

# 커밋 타입 : 커밋 제목 (계정명/저장소명#이슈번호) 형식으로 작성하며 제목은 최대 50글자 정도로만 입력
# 제목을 아랫줄에 작성, 제목 끝에 마침표 금지, 무엇을 했는지 명확하게 작성

kimkyunghun3 committed:
:recycle: CardListButton 메서드 분리 몇 네이밍 변경

# :tada: = 프로젝트 시작 (#)
# :sparkles: = create 파일 생성 시 (#)
# :bulb: = update 코드 및 UI 업데이트, 기능 구현 시 (#)
# :rocket: = release 배포 시 (#)
# :recycle: = refactor 리팩토링 시 (기능은 같은데 코드가 달라지는경우) (#)
# :memo: = 문서 작성 시 (#)
# :art: = 디자인 시스템 관련 내용 (Color / Asset) (#)
# :hammer: = 테스트 코드 작성 (#)
################

# 커밋 타입 : 커밋 제목 (계정명/저장소명#이슈번호) 형식으로 작성하며 제목은 최대 50글자 정도로만 입력
# 제목을 아랫줄에 작성, 제목 끝에 마침표 금지, 무엇을 했는지 명확하게 작성
